### PR TITLE
feat: 重新实现框架对构建工具协议

### PR DESCRIPTION
**Pull request by DDZD20**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/xun082/create-neat/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Test related changes
- [ ] Security fix
- [ ] Dependency update
- [ ] Revert changes
- [ ] Other... Please describe:

## What is the current behavior?

- Closes #
- Related to #

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/core/src/configs/protocol.ts [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
   ENTRY_FILE: "ENTRY_FILE", // 入口文件配置，如：入口文件引入全局 less、scss
   // ……
 };
 
 // 插件对构建工具的协议
-
+exports.pluginToBuildToolProtocol = {};
 // 框架对构建工具的协议
+exports.templateToBuildToolProtocol = {
+  ENTRY_FILE: "ENTRY_FILE", //入口文件配置，比如根据不同框架引入不同方法
```
Comment (uaenaTzx) [inline packages/core/src/configs/protocol.ts:11]: 我感觉通用协议可以维护在一个对象里面，通过 ...XXX 的方式存入每个 protocol 中

File: packages/core/src/models/BaseAPI.ts [hunk 2]
```diff
@@ -3,6 +3,13 @@ import path from "path";
 
 import Generator from "./Generator";
 import PluginToTemplateAPI from "./protocolGenerator/PluginToTemplateAPI";
+import TemplateToBuildToolAPI from "./protocolGenerator/TemplateToBuildToolAPI";
+
+const {
```
Comment (uaenaTzx) [inline packages/core/src/models/BaseAPI.ts:8]: 怎么一会 esm 一会 cjs 的写法

File: packages/core/src/models/protocolGenerator/TemplateToBuildToolAPI.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      return module;
+    } catch (error) {
+      console.error(`Error loading module at ${resolvedPath}:`, error);
+      return null;
+    }
+  }
+  // 入口的先暂时不做
+  // ENTRY_FILE(params) {
+  // }
+  async ADD_CONFIG(params) {
```
Comment (uaenaTzx) [inline packages/core/src/models/protocolGenerator/TemplateToBuildToolAPI.ts:43]: 强烈建议 ts 类型声明一下

File: packages/core/template/template-vue/generator/index.cjs [hunk 4]
```diff
@@ -7,4 +9,17 @@ module.exports = (templateAPI) => {
       "vue-template-compiler": "^2.7.16",
     },
   });
+
+  templateAPI.protocolGenerate({
+    [templateToBuildToolProtocol.ADD_CONFIG]: {
+      params: {
+        content: {
+          buildToolName: 'webpack',  //这里怎么拿选项呢？
```
Comment (uaenaTzx) [inline packages/core/template/template-vue/generator/index.cjs:17]: 不应该在这里传入吧，preset 在 Generator 里面就可以拿到

File: packages/core/src/models/protocolGenerator/TemplateToBuildToolAPI.ts (hunk 3 above)
Comment (DDZD20) [reply packages/core/src/models/protocolGenerator/TemplateToBuildToolAPI.ts:43]: ok

File: packages/core/template/template-vue/generator/index.cjs (hunk 4 above)
Comment (DDZD20) [reply packages/core/template/template-vue/generator/index.cjs:17]: 嗯嗯，这里的 content 现在是准备做一些特殊插件的内容，preset 已经在 baseAPI 拿到并传入 protocols。

File: packages/core/src/configs/protocol.ts (hunk 1 above)
Comment (DDZD20) [reply packages/core/src/configs/protocol.ts]: 已修改

**Pull request closed without merging**